core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -226,6 +228,74 @@ public void cancelOperation() throws UnsupportedOperationException {
         return this.fluxHandle.blockLast();
     }
 
+    /**
+     * Blocks until given {@link OperationStatus} is received or a timeout expires.
+     * @param statusToBlockFor The {@link OperationStatus} to block for and it can be any valid {@link OperationStatus} except {@link OperationStatus#OTHER}.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtil(PollResponse.OperationStatus statusToBlockFor, Duration timeout) {
+        if (statusToBlockFor == null || statusToBlockFor == PollResponse.OperationStatus.OTHER) {
+            throw new IllegalArgumentException("The status input parameter can not be null or OTHER.");
+        }
+        Objects.requireNonNull(timeout, "The timeout input parameter cannot be null.");
+        if (timeout.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for timeout not allowed.");
+        }
+
+        if (!isAutoPollingEnabled()) {
+            setAutoPollingEnabled(true);
+        }
+        return this.fluxHandle.filter(tPollResponse -> matchStatus(tPollResponse, null, statusToBlockFor)).blockFirst(timeout);
+    }
+
+    /**
+     * Blocks until given other operation status is received or a timeout expires.
+     * @param otherStatusToBlockFor The other operation status to block for. {@code Null} and empty string are invalid.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtilOther(String otherStatusToBlockFor, Duration timeout) {
```

Comment:
The more I see `String otherStatus` the stronger I feel about moving the OperationStatus enum class should be moved to the expandable enum, it would simplify our interfaces and make learning the SDK easier.

Reply:
Now implemented ExpandableStringEnum